### PR TITLE
feat: new error types for user.com destination

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/user/transform.js [hunk 1]
```diff
@@ -57,7 +63,7 @@ const identifyResponseBuilder = async (message, destination) => {
 
 const trackResponseBuilder = async (message, destination) => {
   if (!message.event) {
-    throw new CustomError("[ User.com ]:: parameter event is required", 400);
+    throw new InstrumentationError("[User.com]:: parameter event is required");
```
Comment (saikumarrs) [inline src/v0/destinations/user/transform.js:66]: There is no need to add the destination name in the error messages.
```suggestion
    throw new InstrumentationError("parameter event is required");
```

File: src/v0/destinations/user/transform.js [hunk 2]
```diff
@@ -135,16 +139,15 @@ const groupResponseBuilder = async (message, destination) => {
     );
     return responseBuilder(payload, endpoint, method, apiKey);
   }
-  throw new CustomError("[ User.com ] :: No user found with given userId", 400);
+  throw new InstrumentationError(
```
Comment (saikumarrs) [inline src/v0/destinations/user/transform.js:142]: Should it be `NetworkInstrumentationError` as this resulted after contacting the server?

**Review by saikumarrs — COMMENTED** [review]
Good job.
Is there no need to update the router test data?

File: src/v0/destinations/user/transform.js [hunk 3]
```diff
@@ -76,9 +82,8 @@ const trackResponseBuilder = async (message, destination) => {
     return responseBuilder(payload, endpoint, method, apiKey);
   }
 
-  throw new CustomError(
-    "[ User.com ]:: No user found with given lookup field, Track event cannot be completed if there is no valid user",
-    400
+  throw new NetworkInstrumentationError(
+    "[User.com] :: No user found with given lookup field, Track event cannot be completed if there is no valid user"
```
Comment (koladilip) [inline src/v0/destinations/user/transform.js:86]: try not put [User.com] :: these in error messages

Comment (mihir-4116) [reply src/v0/destinations/user/transform.js]: updated

File: src/v0/destinations/user/transform.js (hunk 2 above)
Comment (mihir-4116) [reply src/v0/destinations/user/transform.js]: updated

File: src/v0/destinations/user/transform.js (hunk 1 above)
Comment (mihir-4116) [reply src/v0/destinations/user/transform.js]: updated

**Comment by mihir-4116** [comment]
> Good job. Is there no need to update the router test data?
there is no error in router test cases

**Review by saikumarrs — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**